src/fsharp/service/FSharpCheckerResults.fs

```diff
@@ -1883,6 +1883,19 @@ type FSharpCheckFileResults
                             let symbol = FSharpSymbol.Create(cenv, symbolUse.Item)
                             yield FSharpSymbolUse(scope.TcGlobals, symbolUse.DisplayEnv, symbol, symbolUse.ItemOccurence, symbolUse.Range) |])
 
+    member __.GetAllUsesOfAllSymbolsInFileWithinRange(outerRange: range, ?cancellationToken: CancellationToken) = 
+        threadSafeOp 
+            (fun () -> [| |])
+            (fun scope ->
+                let cenv = scope.SymbolEnv
+                [| 
+                    for symbolUseChunk in scope.ScopeSymbolUses.AllUsesOfSymbols do
+                        for symbolUse in symbolUseChunk do
+                            cancellationToken |> Option.iter (fun ct -> ct.ThrowIfCancellationRequested())
+                            if rangeContainsRange outerRange symbolUse.Range && symbolUse.ItemOccurence <> ItemOccurence.RelatedText then
```

Comment:
Similar to the other pull request, I do wonder if we shouldn't just make `GetAllUsesOfAllSymbolsInFile` return a `seq { ... }`

It might depend on whether we eventually add some kind of indexing, though my experience is that building the index takes as long as any time we save from it.  There may however be ways of building a rough index very quickly, e.g. simply having the tcSink divide things into 16 buckets of  symbols and no more.  (Building an exact hash index via a Dictionary always seemed to be overly accurate indexing for no great benefit.)

Reply:
I will adjust once https://github.com/dotnet/fsharp/pull/10357 is in, since it deals with all the changes that fall out from making it sequential